misskey-websocket/src/client.rs

```diff
@@ -56,16 +57,24 @@ impl Debug for WebSocketClient {
 
 impl WebSocketClient {
     /// Connects to Misskey using WebSocket, and returns [`WebSocketClient`].
-    pub async fn connect(url: Url) -> Result<WebSocketClient> {
-        WebSocketClient::connect_with_config(url, ReconnectConfig::default()).await
+    pub async fn connect<R>(request: R) -> Result<WebSocketClient>
```

Comment:
I don't want to expose `IntoClientRequest` in misskey-websocket's public API. This is because,

1. I want to keep tungstenite hidden from misskey-websocket API
2. `IntoClientRequest` has a kind of pitfall on `http::Request<()>` impl https://docs.rs/tungstenite/latest/tungstenite/client/trait.IntoClientRequest.html and I don't want to explain that in `WebSocketClient` docs
3. `connect_with_config` only uses `headers` and `uri` in it but `http::request::Parts` has some components  other than them and they're implicitly discarded, which is a confusing behavior for users

How about adding (possibly private) method `async fn connect_with_headers_and_config(Url, HeaderMap, ReconnectConfig) -> Result<WebSocketClient>` and wrap this in `connect_with_config` and `connect`?